internal/storage/writer/talaria/talaria.go

```diff
@@ -0,0 +1,62 @@
+package talaria
+
+import (
+	"context"
+	"sync"
+
+	talaria "github.com/kelindar/talaria/client/golang"
+	"github.com/kelindar/talaria/internal/encoding/key"
+	"github.com/kelindar/talaria/internal/monitor/errors"
+)
+
+// Writer to write to TalariaDB
+type Writer struct {
+	lock     sync.Mutex
+	endpoint string
+	client   *talaria.Client
+}
+
+// New initializes a new Talaria writer.
+func New(endpoint string) (*Writer, error) {
+	talaria, err := talaria.Dial(endpoint)
+	if err != nil {
+		return nil, errors.Internal("talaria: unable to connect", err)
+	}
+
+	return &Writer{
+		client:   talaria,
+		endpoint: endpoint,
+	}, nil
+}
+
+// Write will write the ORC data to Talaria
+func (w *Writer) Write(key key.Key, val []byte) error {
+	err := w.tryConnect()
+	if err != nil {
+		return errors.Internal("talaria: unable to connect", err)
+	}
+
+	if err := w.client.IngestORC(context.Background(), val); err != nil {
+		return errors.Internal("talaria: unable to write", err)
+	}
+	return nil
+}
+
+// tryConnect will reconnect to Talaria if needed
+func (w *Writer) tryConnect() error {
```

Comment:
We probably want some way to re-connect if the connection is lost.

Reply:
There's a call for .tryConnect() in Write() at line 34. Is that not enough or am I missing something?

Reply:
tryConnect function only dials when client == nil. Suppose you have set your client, so client is not nil, but the connection is terminated by the server (for example pod shifted, or envoy got restarted), you have to redial.

Checking if it is nil is insufficient, you have to test connection. But doing this on every write as a blocking call might not be ideal as well.

Reply:
Potentially can do a goroutine within the writer that does health check and redial.

Reply:
I think we can start with a blocking connection state check at first and have a blocking re-dial if the connection is not valid. Otherwise, we might potentially lose data.

Reply:
Referred to GRPC status codes here: https://grpc.github.io/grpc/core/md_doc_statuscodes.html

Just a question regarding redialing, if Talaria A is dialing to Talaria B's load balancer via domain name, it doesn't matter if the pods or envoys are restarted since it will try to reconnect whenever the restart is completed right? If the IP changes I don't see how it will be able to dynamically redial to the new IP.